src/components/common/state-machine/state-machine.ts

```diff
@@ -495,13 +499,28 @@ const authStateMachine = createMachine(
       [PATH_NAMES.RESET_PASSWORD_CHECK_EMAIL]: {
         on: {
           [USER_JOURNEY_EVENTS.RESET_PASSWORD_CODE_VERIFIED]: [
-            PATH_NAMES.RESET_PASSWORD,
+            {
+              target: [PATH_NAMES.RESET_PASSWORD_2FA_SMS],
+              cond: "isPasswordResetJourney",
+            },
+            {
+              target: [PATH_NAMES.RESET_PASSWORD],
+            },
           ],
         },
         meta: {
           optionalPaths: [PATH_NAMES.RESET_PASSWORD_RESEND_CODE],
         },
       },
+      [PATH_NAMES.RESET_PASSWORD_2FA_SMS]: {
+        on: {
+          [USER_JOURNEY_EVENTS.MFA_CODE_VERIFIED]: [
+            {
```

Comment:
Also, do we have a test case for this?

Reply:
The condition requiresTwoFactorAuth will be added as part of AUT-1997, as we need AUT-2139 to be merged in order to do this. 2139 will verify the Mfa type of the user and in the implementation of 1997 we will do both the routing to the correct page if the user has 2fa or redirect to the correct page if the user does not require 2fa

Reply:
In the reset-password-check-email-controller.test.ts we have the "should redirect to check_phone if code entered is correct and feature flag is turned on" which verifies if it redirects to the correct page if the feature flag is on, if that's the test case you are referring to.

Reply:
The query parameters have not been added, thank you!

Reply:
Thanks for the explanation and looking at the query params.

In terms of the test you referenced, that looks to me like it's testing the change you made above this, since it's testing for us ending up on `/reset-password-2fa-sms`, whereas this condition starts from there and ends on `/reset-password`, so looking for a test for that.

Reply:
Oh, I understand now, thank you! Yes, we also need to add a test for this.